src/app.rs

```diff
@@ -223,7 +222,7 @@ impl App {
             for (idx, c) in text.chars().enumerate() {
                 if running.lock().unwrap().force_stop() {
                     // Print the rest of the string all at once.
-                    buffer.lock().unwrap().last_mut().unwrap().text += &text[idx..text.len()];
+                    Self::add_to_buf(buffer, &text[idx..text.len()]);
                     break;
                 }
                 buffer.lock().unwrap().last_mut().unwrap().text.push(c);
```

Comment:
This one can be replaces as well with `add_to_buf()`, i.e., with `&text[idx..idx]` syntax or creating a separate function for chars.